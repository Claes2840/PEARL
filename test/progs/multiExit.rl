init: entry
      if dyn goto fst else snd

fst: from init
     elim ^= 'a
     goto stop

snd: from init
     elim ^= 'b
     goto stop

stop: fi dyn from fst else snd
      exit