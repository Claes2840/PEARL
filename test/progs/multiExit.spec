// None needed
elim1 = 'nil