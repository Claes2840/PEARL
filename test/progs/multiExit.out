(dyn) -> (dyn elim)

init_1:
	entry
	if (dyn = '5)
		goto fst_1_2
		else snd_1_2

fst_1_2:
	from init_1
	assert((dyn = '5))
	dyn += '1
	elim ^= '1
	goto stop_1_x1_2_3

stop_1_x1_2_3:
	fi (elim = '1)
		from fst_1_2
		else snd_1_2
	exit

snd_1_2:
	from init_1
	assert(!((dyn = '5)))
	dyn += '2
	elim ^= '2
	goto stop_1_x1_2_3
